apps-script/incident-response/ChatApp.gs

```diff
@@ -0,0 +1,169 @@
+ *
+ * This app only responds to one kind of dialog (Close Incident).
+ *
+ * @param {Object} event the event object from Google Chat
+ */
+function onCardClick(event) {
+  if (event.isDialogEvent) {
+    if (event.dialogEventType == 'SUBMIT_DIALOG') {
+      return processSubmitDialog_(event);
+    }
+    return {
+      actionResponse: {
+        type: "DIALOG",
+        dialogAction: {
+          actionStatus: "OK"
+        }
+      }
+    };
+  }
+}
+
+/**
+ * Responds to a MESSAGE event with a Slash command in Google Chat.
+ *
+ * This app only responds to a slash command with the ID 1 ("/closeIncident")
+ * by returning a Dialog.
+ *
+ * @param {Object} event the event object from Google Chat
+ */
+function processSlashCommand_(event) {
+  if (event.message.slashCommand.commandId != CLOSE_INCIDENT_COMMAND_ID) {
+    return { "text": "Command not recognized. Use the command `/closeIncident` to close the incident managed by this space." };
+  }
+  return {
+    actionResponse: {
+      type: "DIALOG",
+      dialogAction: {
+        dialog: {
+          body: {
+            sections: [
+              {
+                header: "Close Incident",
+                widgets: [
+                  {
+                    textInput: {
+                      label: "Please describe the incident resolution",
+                      type: "MULTIPLE_LINE",
+                      name: "description"
+                    }
+                  },
+                  {
+                    buttonList: {
+                      buttons: [
+                        {
+                          text: "Close Incident",
+                          onClick: {
+                            action: {
+                              function: "closeIncident"
+                            }
+                          }
+                        }
+                      ]
+                    }
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      }
+    }
+  };
+}
+
+/**
+ * Responds to a CARD_CLICKED event with a Dialog submission in Google Chat.
+ *
+ * This app only responds to one kind of dialog (Close Incident).
+ * It creates a Doc with a summary of the incident information and posts a message
+ * to the space with a link to the Doc.
+ *
+ * @param {Object} event the event object from Google Chat
+ */
+function processSubmitDialog_(event) {
+  const resolution = event.common.formInputs.description[""].stringInputs.value[0];
+  const chatHistory = getChatHistory_(event.space.name);
+  const chatSummary = summarizeChatHistory_(chatHistory);
+  const docUrl = createDoc_(event.space.displayName, resolution, chatHistory, chatSummary);
+  return {
+    actionResponse: {
+      type: "NEW_MESSAGE",
+    },
+    text: `Incident closed with the following resolution: ${resolution}\n\nHere is the automatically generated post-mortem:\n${docUrl}`
+  };
+}
+
+/**
+ * Lists all the messages in the Chat space.
```

Comment:
Comment doesn't really match. Consider naming this function differently, `concatenateAllSpaceMessages` or similar.

Reply:
Done.